src/calendar/MonthPicker.ts

```diff
@@ -0,0 +1,300 @@
+ */
+export interface MonthPickerProperties extends ThemeableProperties {
+	labelId?: string;
+	labels: CalendarMessages;
+	month: number;
+	monthNames: { short: string; long: string; }[];
+	open?: boolean;
+	renderMonthLabel?(month: number, year: number): string;
+	year: number;
+	onRequestOpen?(): void;
+	onRequestClose?(): void;
+	onRequestMonthChange?(month: number): void;
+	onRequestYearChange?(year: number): void;
+};
+
+export const MonthPickerBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export default class MonthPicker extends MonthPickerBase<MonthPickerProperties> {
+	private _buttonId = uuid();
+	private _dialogId = uuid();
+	private _labelId = uuid();
+	private _yearSpinnerId = uuid();
+	private _radiosName = uuid();
+
+	private _callTriggerFocus = false;
+	private _callPopupFocus = false;
+
+	// move focus when opening/closing the popup
+	protected onElementUpdated(element: HTMLElement, key: string) {
+		// TODO: find a way to call focus on button widget
+
+		// popup
+		if (key === 'year-spinner') {
+			const { open } = this.properties;
+			if (open && this._callPopupFocus) {
+				element.focus();
+				this._callPopupFocus = false;
+			}
+		}
+	}
+
+	private _closePopup() {
+		const { onRequestClose } = this.properties;
+
+		this._callTriggerFocus = true;
+		onRequestClose && onRequestClose();
+	}
+
+	private _openPopup() {
+		const { onRequestOpen } = this.properties;
+
+		this._callPopupFocus = true;
+		onRequestOpen && onRequestOpen();
+	}
+
+	private _decreaseYear() {
+		const {
+			year,
+			onRequestYearChange
+		} = this.properties;
+
+		onRequestYearChange && onRequestYearChange(year - 1);
+	}
+
+	private _increaseYear() {
+		const {
+			year,
+			onRequestYearChange
+		} = this.properties;
+
+		onRequestYearChange && onRequestYearChange(year + 1);
+	}
+
+	private _onButtonClick() {
+		const {
+			open = false
+		} = this.properties;
+
+		open ? this._closePopup() : this._openPopup();
+	}
+
+	private _onMonthDecrease() {
+		const {
+			month,
+			year,
+			onRequestMonthChange,
+			onRequestYearChange
+		} = this.properties;
+
+		if (month === 0) {
+			onRequestMonthChange && onRequestMonthChange(11);
+			onRequestYearChange && onRequestYearChange(year - 1);
+		}
+		else {
+			onRequestMonthChange && onRequestMonthChange(month - 1);
+		}
+	}
+
+	private _onMonthIncrease() {
+		const {
+			month,
+			year,
+			onRequestMonthChange,
+			onRequestYearChange
+		} = this.properties;
+
+		if (month === 11) {
+			onRequestMonthChange && onRequestMonthChange(0);
+			onRequestYearChange && onRequestYearChange(year + 1);
+		}
+		else {
+			onRequestMonthChange && onRequestMonthChange(month + 1);
+		}
+	}
+
+	private _onPopupKeyDown(event: KeyboardEvent) {
+		// close popup on escape, or if a value is selected with enter/space
+		if (
+			event.which === Keys.Escape ||
+			event.which === Keys.Enter ||
+			event.which === Keys.Space
+		) {
+			this._closePopup();
+		}
+	}
+
+	private _onRadioChange(event: Event) {
```

Comment:
I think that there is a `TargetedEventObject` in `widget-core` that you could use like `TargetedEventObject<HTMLInputElement>`